### PR TITLE
Add --target to bud command

### PR DESCRIPTION
**Pull request by TomSweeneyRedHat**
Signed-off-by: TomSweeneyRedHat <tsweeney@redhat.com>

Add the --target option to the bud command.  This allows the
user to specify the last stage to build in a multi stage
Dockerfile.

Addresses #632

Tests:
```
cat ./bud/target/Dockerfile
FROM ubuntu:latest
RUN touch /1

FROM alpine:latest AS mytarget
RUN touch /2

FROM busybox:latest AS mytarget2
RUN touch /3

buildah bud --debug=false -t tom --target mytarget ./bud/target .
STEP 1: FROM ubuntu:latest
STEP 2: RUN touch /1
STEP 3: FROM alpine:latest AS mytarget
STEP 4: RUN touch /2
STEP 5: COMMIT containers-storage:[overlay@/var/lib/containers/storage+/var/run/containers/storage]localhost/tom:latest
Getting image source signatures
Skipping blob 503e53e365f3 (already present): 5.52 MiB / 5.52 MiB [=========] 0s
Copying blob 66e5f29a7649: 3.50 KiB / 3.50 KiB [============================] 0s
Copying config e72f1fa3d72d: 704 B / 704 B [================================] 0s
Writing manifest to image destination
Storing signatures
--> e72f1fa3d72ddd3f23acb22a059ecce33dad571433223389e3ce92a5fd9ebae5
STEP 6: COMMIT containers-storage:[overlay@/var/lib/containers/storage+/var/run/containers/storage]localhost/tom:latest
Getting image source signatures
Skipping blob 503e53e365f3 (already present): 5.52 MiB / 5.52 MiB [=========] 0s
Skipping blob 66e5f29a7649 (already present): 3.50 KiB / 3.50 KiB [=========] 0s
Copying config e72620d8efe7: 704 B / 704 B [================================] 0s
Writing manifest to image destination
Storing signatures
--> e72620d8efe764178d1352dfb3a9a773794309ee9e879e17d3803b18553f5025

buildah images
IMAGE NAME                                               IMAGE TAG            IMAGE ID             CREATED AT             SIZE
docker.io/library/ubuntu                                 latest               20bb25d32758         Jan 22, 2019 17:41     90 MB
docker.io/library/alpine                                 latest               caf27325b298         Jan 30, 2019 17:19     5.8 MB
localhost/tom                                            latest               993ee7ded616         Feb 5, 2019 13:49      5.8 MB
```

Signed-off-by: TomSweeneyRedHat <tsweeney@redhat.com>

### CODE REVIEW

**Comment by mheon** [comment]
SGTM

File: imagebuildah/build.go [hunk 1]
```diff
@@ -1441,6 +1443,13 @@ func BuildDockerfiles(ctx context.Context, store storage.Store, options BuildOpt
 	if err != nil {
 		return "", nil, errors.Wrap(err, "error reading multiple stages")
 	}
+	if (options.Target != "") {
+		stagesTargeted, ok := stages.ByTarget(options.Target)
+		if !ok {
+			return "", nil, errors.Wrapf(err, "The target %q was not found in the provided Dockerfile", options.Target)
```
[the remark this answers is not in the document]
Comment (TomSweeneyRedHat) [reply imagebuildah/build.go]: Yep, cut/paste error from imagebuilder, I'll touch it up.

**Comment by TomSweeneyRedHat** [comment]
Repushed with gofmt tended to and @rhatdan's comment addressed.

File: tests/bud/target/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,5 @@
+FROM fedora:latest
+RUN touch /foo
+
+FROM alpine:latest AS mytarget
```
Comment (umohnani8) [inline tests/bud/target/Dockerfile]: Maybe add one more stage after this to ensure that it is skipped when --target it set to mytarget. To ensure we don't accidentally regress in future code changes.

**Comment by umohnani8** [comment]
One comment, but LGTM

**Comment by vrothberg** [comment]
Moby seems to behave differently. Currently, we only execute the target but Moby is *stopping* at the specified target and executes all previous stages.

```Dockerfile
FROM alpine:latest as one        
RUN touch ~/alpine.txt           
                                 
FROM centos:latest as centos     
RUN touch ~/centos.txt           
                                 
FROM fedora:latest as fedora     
RUN touch ~/fedora.txt           
```

Running `docker build --target centos` on the upper Dockerfile:
```bash
Sending build context to Docker daemon  2.048kB
Step 1/4 : FROM alpine:latest as one
latest: Pulling from library/alpine
6c40cc604d8e: Pull complete
Digest: sha256:b3dbf31b77fd99d9c08f780ce6f5282aba076d70a513a8be859d8d3a4d0c92b8      
Status: Downloaded newer image for alpine:latest
 ---> caf27325b298
Step 2/4 : RUN touch ~/alpine.txt
 ---> Running in 923b3b87dfd6
Removing intermediate container 923b3b87dfd6
 ---> 544aae7ed2c4
Step 3/4 : FROM centos:latest as centos
latest: Pulling from library/centos
a02a4930cb5d: Pull complete
Digest: sha256:184e5f35598e333bfa7de10d8fb1cebb5ee4df5bc0f970bf2b1e7c7345136426      
Status: Downloaded newer image for centos:latest
 ---> 1e1148e4cc2c
Step 4/4 : RUN touch ~/centos.txt
 ---> Running in c46c76e02eca
Removing intermediate container c46c76e02eca
 ---> ef6c384fa716
Successfully built ef6c384fa716
```

**Comment by TomSweeneyRedHat** [comment]
I've had to change this to WIP until https://github.com/openshift/imagebuilder/pull/121 is merged and then I can vendor it here.  Test should fail until that happens as the new ThroughTarget() function is not yet available.

**Comment by TomSweeneyRedHat** [comment]
Since the last push, I've updated the vendor for imagebuilder and made one small change in wording in the man page.  Fingers crossed, this is gtg once the tests pass.

**Comment by rhatdan** [comment]
I will leave this to @vrothberg to approve.

**Comment by vrothberg** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**